app/models/async_transaction/vet360/base.rb

```diff
@@ -2,6 +2,32 @@
 
 module AsyncTransaction
   module Vet360
-    class Base < AsyncTransaction::Base; end
+    class Base < AsyncTransaction::Base;
+
+      def self.refresh_transaction_status(user, tx_id = nil)
+
+        transaction_record = Base.find_by(user_uuid: user.uuid, transaction_id: tx_id)
+
+        return unless transaction_record
```

Comment:
Let's raise an error here, because it is an exceptional situation to not find a record the frontend thinks should exist.